### ISSUE TITLE
Scaletesting: [IAM] 12k synthetic users and 8k teams (50 users & 50 repos per team)

### ISSUE THREAD

**Issue by jhchabran** [sourcegraph/sourcegraph#43052]
/cc @sourcegraph/iam

**Comment by jhchabran** [comment]
Blocked by #43044

**Comment by sanderginn** [comment]
- [Environment details](https://sourcegraph.slack.com/archives/C03GMM0F2BA/p1665658367720609) from https://github.com/sourcegraph/accounts/issues/1

**Comment by kopancek** [comment]
As for the representation of permissions on the repositories, I would like to get it to this state:
50% of repositories - permissions are listed for < 10 teams and < 5 individual users
40% of repositories - permissions are listed for < 10 users and no teams/orgs
10% of repositores - permissions are listed for > 100, but < 500 individual users

We also need some repositories (1%) to have orgs assigned instead of teams - orgs would be much larger than teams, e.g. > 1000 users.

**Comment by sanderginn** [comment]
Closing this as we are at a first milestone. Further work to support larger scales will be added to a new ticket.

Currently on the scaletesting GHE instance:
- 40k users
- 5k teams
- 1 main org + 5 sub-orgs
- 200k repositories:
  - 4750 teams each have 27 or 28 repositories assigned 
  - 200 teams each have 78 repositories assigned
  - 50 teams each have 985 repositories assigned
  - Each sub-org has 400 repositories assigned and 2000 users in it, with no team or user repo memberships
  - 1000 repositories each have 3 users directly assigned (and are only visible to those users)
